platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/IssDetector.java

```diff
@@ -548,4 +549,14 @@ private static void writeSkippedLogCount(@NonNull final StringBuilder sb, final
                     .append("seconds.");
         }
     }
+
+    /**
+     * Checks equality of hashes when they are stored in different formats.
+     */
+    private static boolean hashEquals(@Nullable final Hash a, @NonNull final Bytes b) {
+        if (a == null && b.length() == 0) {
+            return true;
+        }
+        return a != null && a.equalBytes(b);
```

Comment:
generally I have lower standards for private methods. Since this method is very specific to this class, I would keep it mostly as is, its not a general purpose method. I renamed the parameters though.